code/utility/src/palindrome/palindrome_check/palindrome.nim

```diff
@@ -0,0 +1,14 @@
+func is_palindrome(str: string): bool =
+    let length = str.len - 1
+    for i in 0..(length mod 2):
+      if str[i] != str[length - i]:
+        return false
+    true
+  
+func test() =
+    assert("madam".is_palindrome)
+    assert(not is_palindrome("nimrod"))
```

Comment:
Please use UFCS here too. Also I would make this is a `main` function, called `when isMainModule`. Then this should be good to merge. You may also consider adding recursive and string-reversing implementations of this algorithm in this same file.